src/transformers/training_args.py

```diff
@@ -1194,6 +1194,9 @@ def _setup_devices(self) -> "torch.device":
         if self.no_cuda:
             device = torch.device("cpu")
             self._n_gpu = 0
+            os.environ["RANK"] = str(os.environ.get("PMI_RANK", 0))
+            os.environ["WORLD_SIZE"] = str(os.environ.get("PMI_SIZE", 1))
+            self.local_rank = int(os.environ.get("MPI_LOCALRANKID", -1))
```

Comment:
This will break any distributed training on CPU not using MPI (you are overwriting global env variables that might already have been set), so we can't accept this as a fix.

Reply:
sgugger, thanks for your comment, I upload a new commit to fix the global env variable overwriting issue you mentioned. the issue is when you miprun the example "without accelerate", the self.local_rank = -1, so that distribute backend is not initiated at all.